### PR TITLE
Add a new `scaffold` command to the CLI

### PR DESCRIPTION
**Pull request by jakedahn**
This PR introduces a new `scaffold` command to the Replicate CLI.

You can use this command to bootstrap a completely functional development environment from a given prediction ID or URL.

To start, support is limited to javascript via the https://github.com/replicate/node-starter repo and python via the https://github.com/replicate/python-starter repo.

# Usage

Scaffold via prediction URL

* `replicate scaffold https://replicate.com/p/z5kvo4tb6mhb7xnpxpjpfxlzxe /path/to/repo` (default uses node)
* `replicate scaffold --template node https://replicate.com/p/z5kvo4tb6mhb7xnpxpjpfxlzxe /path/to/repo` (node)
* `replicate scaffold --template python https://replicate.com/p/z5kvo4tb6mhb7xnpxpjpfxlzxe /path/to/repo` (python)

Scaffold via prediction ID

* `replicate scaffold z5kvo4tb6mhb7xnpxpjpfxlzxe /path/to/repo` (default uses node)
* `replicate scaffold --template node z5kvo4tb6mhb7xnpxpjpfxlzxe /path/to/repo` (node)
* `replicate scaffold --template python z5kvo4tb6mhb7xnpxpjpfxlzxe /path/to/repo` (python)

### CODE REVIEW

**Comment by zeke** [comment]
Should we also support this other format of prediction URL?

```
https://replicate.com/stability-ai/stable-diffusion?prediction=emeqxllb6c4uxjy2dbun6yk62u
```

**Review by mattt — COMMENTED** [review]
This is great, @jakedahn. Thank you so much for working on this. I left some specific code review suggestions for your consideration.

Some more high-level feedback:

- I'm not sure `clone` is the right spelling for this command. I slightly prefer `scaffold`, but maybe there's another option `bootstrap`? `project init`?
  - If `project init`, then maybe the prediction is optional, and you just use a default example, like SDXL.

- Looking what's in the [starter repos](https://github.com/replicate/python-starter), I think it might make sense to embed them in the CLI itself. Go has a nice affordance for [embedding](https://pkg.go.dev/embed) source files as string constants to make that convenient.
  - If the concern is future-proofing, then we could put the template files in the CLI repo and fetch the latest versions.

**Comment by bfirsh** [comment]
Agreed clone feels a bit funny. Not sure why. Perhaps because it's a bit of a different intent from Git? Perhaps because having it as a top-level command implies it's a primary action of Replicate the product? Run, models, deployments... clone?

`init` is a pattern we have in Cog. Scaffold and bootstrap both seem pretty good too. No strong feelings overall.

File: internal/cmd/clone/root.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+	"github.com/replicate/replicate-go"
+	"github.com/spf13/cobra"
+)
+
+var RootCmd = &cobra.Command{
+	Use:   "clone [prediction ID or URL] [output path]",
+	Short: "Setup a new local development environment from a prediction",
+	Args:  cobra.ExactArgs(2),
+	Run: func(cmd *cobra.Command, args []string) {
```
[the remark this answers is not in the document]
Comment (jakedahn) [reply internal/cmd/clone/root.go]: 🤯 so much better

**Comment by zeke** [comment]
> Looking what's in the [starter repos](https://github.com/replicate/python-starter), I think it might make sense to embed them in the CLI itself.

I tend to agree.

**Comment by jakedahn** [comment]
> > Looking what's in the [starter repos](https://github.com/replicate/python-starter), I think it might make sense to embed them in the CLI itself.
> 
> I tend to agree.

I disagree here.

I think it's a strange experience to be like "oh I would really love to add a starter template for elixir", and then be forced to vendor everything inside of a golang codebase.

The hope behind using actual git repos here is that in a future iteration, someone could say `replicate init --template https://github.com/cbh123/replicate-elixir.git <prediction id> /path/to/checkout`. Allowing anybody the ability to create their own starter templates that hook into the same cli.

This is not implemented in this PR as-is, but this is a future I think is worth heading towards.

**Comment by jakedahn** [comment]
I've renamed everything to `replicate init` 🙌

**Comment by mattt** [comment]
> The hope behind using actual git repos here is that in a future iteration, someone could say `replicate init --template https://github.com/cbh123/replicate-elixir.git <prediction id> /path/to/checkout`. Allowing anybody the ability to create their own starter templates that hook into the same cli.

@jakedahn 100% with you. I think we can get the best of both worlds by locating built-in templates (`python`, `node`) in CLI repo, and allowing user-provided URLs to repos. 

You could even implement this internally by routing aliases like `python` to the full path in the CLI repo on GitHub. You'd just need to support paths to subdirectories.

**Comment by mattt** [comment]
@jakedahn Thanks again for your great work on this. I'm excited to have this in the CLI.

I was thinking about this some more the other day, and came up with a slightly different interface:

```bash
# Generate from a prediction
replicate prediction scaffold z5kvo4tb6mhb7xnpxpjpfxlzxe # also supports URLs


# Generate from a model's default example
replicate model scaffold stability-ai/sdxl # also supports URLs, optional version


# Top-level alias that automatically does the right thing
replicate scaffold z5kvo4tb6mhb7xnpxpjpfxlzxe # prediction ID
replicate scaffold stability-ai/sdxl # model name
replicate scaffold # default project template 
```

In the interest of shipping fast and iterating, the minimal change would be to rename `init` to `scaffold`. From there, we could fill in the respective subcommands and update the top-level command to support no argument and a model argument.

How does that sound?

**Pull request merged**